ci/src/main/java/group10/GithubController.java

```diff
@@ -3,21 +3,28 @@
 import spark.Request;
 import spark.Response;
 
+import java.io.File;
+import java.io.IOException;
+
+import org.json.JSONException;
 import org.json.JSONObject;
 
 import group10.util.JsonUtil;
 
 public class GithubController {
     private static JsonUtil json = new JsonUtil();
+    private static CIServer ciServer = new CIServer();
 
     /**
-     * Handle the Github post event: get info, clone
-     * run tests and return response.
-     * @param request from github
+     * Handle the Github post event: get info, clone run tests and return response.
+     * 
+     * @param request  from github
      * @param response to send back
      * @return
+     * @throws IOException
+     * @throws JSONException
      */
-    public static String handlePost(Request request, Response response) {
+    public static String handlePost(Request request, Response response) throws JSONException, IOException {
         // turn data string into a map structure
```

Comment:
Should probably have try-catch for the methods throwing errors and log this, otherwise, the web server will crash and stop responding if an error occurs (I think).

Reply:
Fixed